neuro_flow/context.py

```diff
@@ -678,6 +683,44 @@ async def setup_strategy_ctx(
     return StrategyCtx(fail_fast=fail_fast, max_parallel=max_parallel)
 
 
+async def setup_matrix(ast_matrix: Optional[ast.Matrix]) -> Sequence[MatrixCtx]:
+    if ast_matrix is None:
+        return [{}]
+    # Init
+    products = []
+    for k, lst in ast_matrix.products.items():
+        lst2 = [{k: await i.eval(EMPTY_ROOT)} for i in lst]
+        products.append(lst2)
+    matrices = []
+    for row in itertools.product(*products):
+        dct: Dict[str, LiteralT] = {}
+        for elem in row:
+            dct.update(elem)
+        matrices.append(dct)
+    # Include
+    for inc_spec in ast_matrix.include:
```

Comment:
I believe the inclusion should be processed after exclusion.
At least GitHub has such logic, please see the note at the very end of the chapter about matrix: https://docs.github.com/en/free-pro-team@latest/actions/reference/workflow-syntax-for-github-actions#jobsjob_idstrategymatrix

Reply:
Sorry, seems like I add this mistake in refactoring PR. Thanks!
I'll fix this and add some unit test.